### PR TITLE
Store tunneltoken and notunnel data in database.

### LINKED ISSUE #137: Use SQLite instead of a flat file to store tunneltoken and notoken

### PR DESCRIPTION
**Pull request by mrstegeman**
Fixes #137

### CODE REVIEW

File: src/controllers/settings_controller.js [hunk 1]
```diff
@@ -160,21 +161,31 @@ SettingsController.post('/subscribe', function (request, response) {
 });
 
 SettingsController.post('/skiptunnel', function (request, response) {
```
Comment (hobinjk) [inline src/controllers/settings_controller.js]: Ideally SettingsController would be a PromiseRouter then you could write this with async/await and be more explicit about the lifetime of the route handler.

File: src/controllers/users_controller.js [hunk 2]
```diff
@@ -61,23 +61,30 @@ UsersController.post('/', async (request, response) => {
...
-        // settings page to allow him to retry
-        throw e;
-    })
-    .then(function(res) {
-        return res.text();
-    })
-    .then(function() {
-        console.log('Online account created.');
+  if (config.get('ssltunnel.enabled')) {
+    Settings.get('tunneltoken').then(function(result) {
```
Comment (hobinjk) [inline src/controllers/users_controller.js]: This and the fetch above can both be simplified using async/await

File: src/app.js [hunk 3]
```diff
@@ -32,6 +33,25 @@ require('./app-instance');
...
+  Settings.set('tunneltoken', token).then(function() {
+    fs.unlinkSync('tunneltoken');
+  }).catch(function(e) {
+    throw e;
+  });
+}
+
+// Move the notunnel setting into the database
+if (fs.existsSync('notunnel')) {
+  Settings.set('notunnel', false).then(function() {
```
[the remark this answers is not in the document]
Comment (mrstegeman) [reply src/app.js]: ... yes. Good catch.

**Comment by hobinjk** [comment]
Something's going wrong with attempting to register a new Gateway with these changes. I moved `*.pem` and `tunneltoken` into another directory, restarted my server, and now I'm getting a network error attempting to create a new domain

File: src/app.js [hunk 4]
```diff
@@ -32,17 +33,34 @@ require('./app-instance');
...
+let httpsServer = createHttpsServer();
 let httpsApp = null;
 
-function startHttpsGateway() {
-  let port = config.get('ports.https');
-  const cliOptions = getOptions();
-  if (typeof cliOptions.port === 'number') {
-    port = cliOptions.port;
+function createHttpsServer() {
+  if (!fs.existsSync('privatekey.pem') || !fs.existsSync('certificate.pem')) {
```
Comment (hobinjk) [inline src/app.js]: Could be replaced with `TunnelService.hasCertificates`

File: src/app.js [hunk 5]
```diff
@@ -53,7 +71,16 @@ function startHttpsGateway() {
...
+  return https.createServer(options);
+}
+
+function startHttpsGateway() {
+  let port = config.get('ports.https');
+  const cliOptions = getOptions();
+  if (typeof cliOptions.port === 'number') {
+    port = cliOptions.port;
+  }
+
```
Comment (hobinjk) [inline src/app.js]: Should include:
```javascript
if (!httpsServer) {
  httpsServer = createHttpsServer();
}
```

**Pull request merged**